File: VB/HowToUseTreeListEditor.Module/Issue.vb
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Base.General

Namespace HowToUseTreeListEditor.Module

    <DefaultClassOptions>
    Public Class Issue
        Inherits BaseObject
        Implements ICategorizedItem

        Private categoryWithIssues As CategoryWithIssues

        Private subjectField As String

        Private descriptionField As String

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Sub New(ByVal session As Session, ByVal subject As String)
            MyBase.New(session)
            subjectField = subject
        End Sub

        <Association("CategoryWithIssues-Issues")>
        Public Property CategoryProp As CategoryWithIssues
            Get
                Return categoryWithIssues
            End Get

            Set(ByVal value As CategoryWithIssues)
                SetPropertyValue("Category", categoryWithIssues, value)
            End Set
        End Property

        Public Property Subject As String
            Get
                Return subjectField
            End Get

            Set(ByVal value As String)
                SetPropertyValue("Subject", subjectField, value)
            End Set
        End Property

        Public Property Description As String
            Get
                Return descriptionField
            End Get

            Set(ByVal value As String)
                SetPropertyValue("Description", descriptionField, value)
            End Set
        End Property

        Private Property Category As ITreeNode Implements ICategorizedItem.Category
            Get
                Return CategoryProp
            End Get

            Set(ByVal value As ITreeNode)
                CategoryProp = CType(value, CategoryWithIssues)
            End Set
        End Property
    End Class
End Namespace

File: CS/HowToUseTreeListEditor.Module/Issue.cs
using System;

using DevExpress.Xpo;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;
using DevExpress.Persistent.Base.General;

namespace HowToUseTreeListEditor.Module {
    [DefaultClassOptions]
    public class Issue : BaseObject, ICategorizedItem {
        private CategoryWithIssues categoryWithIssues;
        private string subject;
        private string description;
        public Issue(Session session) : base(session) { }
        public Issue(Session session, string subject)
            : base(session) {
            this.subject = subject;
        }
        [Association("CategoryWithIssues-Issues")]
        public CategoryWithIssues Category {
            get {
                return categoryWithIssues;
            }
            set {
                SetPropertyValue("Category", ref categoryWithIssues, value);
            }
        }
        public string Subject {
            get {
                return subject;
            }
            set {
                SetPropertyValue("Subject", ref subject, value);
            }
        }
        public string Description {
            get {
                return description;
            }
            set {
                SetPropertyValue("Description", ref description, value);
            }
        }
        ITreeNode ICategorizedItem.Category {
            get {
                return Category;
            }
            set {
                Category = (CategoryWithIssues)value;
            }
        }
    }


}
